src/audio_unit/delay.rs

```diff
@@ -57,12 +62,29 @@ impl Delay {
         }
     }
 
-    fn delay_num_samples(stream_config: &StreamConfig, delay_ms: u32) -> usize {
+    fn delay_num_samples(stream_config: &StreamConfig, delay_ms: DelayMs) -> usize {
         let delay_num_frames = (delay_ms as f32 / 1_000.0) * stream_config.sample_rate.0 as f32;
         delay_num_frames as usize * stream_config.channels as usize
     }
 
-    fn set_delay_ms(&mut self, delay_ms: u32) -> Result<()> {
+    fn process_messages(&mut self) -> Result<()> {
+        let messages: Vec<_> = self.messages.try_iter().collect();
+        for message in messages {
+            self.process_message(message)?;
+        }
+
+        Ok(())
+    }
+
+    fn process_message(&mut self, message: Message) -> Result<()> {
+        match message {
+            SetDelay(delay) => self.set_delay_ms(delay)?,
+        };
+
+        Ok(())
+    }
```

Comment:
This feels a bit boilerplate-y, but also nice and explicit, and should scale up to much more complex audio units (if that is a thing).